### ISSUE TITLE
Validation loss unstable

### ISSUE THREAD

**Issue by genz10** [erickrf/autoencoder#6]
Hi, 

I've been run training autoencoder for a couple minutes and find validation loss was unstable, while average batch loss keep smaller.

```
INFO:root:68 epochs, 480 batches	Avg batch loss: 0.069513	Validation loss: 2.792187
INFO:root:69 epochs, 490 batches	Avg batch loss: 0.072354	Validation loss: 3.593376
INFO:root:71 epochs, 500 batches	Avg batch loss: 0.066709	Validation loss: 2.902308
INFO:root:72 epochs, 510 batches	Avg batch loss: 0.068916	Validation loss: 2.456438
INFO:root:74 epochs, 520 batches	Avg batch loss: 0.063203	Validation loss: 3.335535
INFO:root:75 epochs, 530 batches	Avg batch loss: 0.061095	Validation loss: 2.947956
INFO:root:77 epochs, 540 batches	Avg batch loss: 0.053819	Validation loss: 2.774727
INFO:root:78 epochs, 550 batches	Avg batch loss: 0.049284	Validation loss: 2.505195
INFO:root:79 epochs, 560 batches	Avg batch loss: 0.048512	Validation loss: 2.992307
INFO:root:81 epochs, 570 batches	Avg batch loss: 0.046487	Validation loss: 2.499180
INFO:root:82 epochs, 580 batches	Avg batch loss: 0.047605	Validation loss: 2.706982
INFO:root:84 epochs, 590 batches	Avg batch loss: 0.043647	Validation loss: 3.071584
INFO:root:85 epochs, 600 batches	Avg batch loss: 0.043592	Validation loss: 2.697872
INFO:root:87 epochs, 610 batches	Avg batch loss: 0.040263	Validation loss: 2.809274
INFO:root:88 epochs, 620 batches	Avg batch loss: 0.036155	Validation loss: 3.256417
INFO:root:89 epochs, 630 batches	Avg batch loss: 0.038276	Validation loss: 3.183321
INFO:root:91 epochs, 640 batches	Avg batch loss: 0.034546	Validation loss: 3.526584
INFO:root:92 epochs, 650 batches	Avg batch loss: 0.035519	Validation loss: 3.173420
INFO:root:94 epochs, 660 batches	Avg batch loss: 0.033806	Validation loss: 2.976058
INFO:root:95 epochs, 670 batches	Avg batch loss: 0.035120	Validation loss: 4.046538
INFO:root:97 epochs, 680 batches	Avg batch loss: 0.030202	Validation loss: 2.641236
```

What actually cause this problem ?

Thanks !

**Comment by erickrf** [comment]
Considering that your epochs fit in 10 batches, it looks like you are using too few training data and the model can't learn properly.

**Comment by genz10** [comment]
@erickrf 

Thanks for your reply. I'm using small data for proofing concept. CMIIW, I expect with small dataset, it will be overfit, and when running with validation data set will produce small loss value.

**Comment by genz10** [comment]
I've try full data set, the validation loss is increasing

```
INFO:root:14 epochs, 100 batches	Avg batch loss: 2.831865	Validation loss: 5.789834
INFO:root:21 epochs, 150 batches	Avg batch loss: 1.753031	Validation loss: 6.345498
INFO:root:28 epochs, 200 batches	Avg batch loss: 1.083488	Validation loss: 6.580184
INFO:root:35 epochs, 250 batches	Avg batch loss: 0.647666	Validation loss: 6.866635
INFO:root:42 epochs, 300 batches	Avg batch loss: 0.392439	Validation loss: 7.370073
```

**Comment by erickrf** [comment]
It still seems to be very few data. The training loss values are decreasing, which indicates it is overfitting the data.
